// apeman-react-clock/shim/node/ap_analog_clock_hand.js
/**
 * A hand for analog clock.
 * @class ApAnalogClockHand
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAnalogClockHand */
var ApAnalogClockHand = _react2.default.createClass({
  displayName: 'ApAnalogClockHand',

  propTypes: {
    angle: _react.PropTypes.number,
    width: _react.PropTypes.number,
    heightRate: _react.PropTypes.number
  },
  getDefaultProps: function getDefaultProps() {
    return {
      angle: 0,
      width: 24,
      heightRate: 1
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;


    var className = props.className,
        angle = props.angle,
        heightRate = props.heightRate,
        width = props.width;

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)("ap-analog-clock-hand", className),
        style: { transform: 'rotate(' + angle + 'deg)' } },
      _react2.default.createElement(
        'div',
        { className: 'ap-analog-clock-hand-bar-container',
          style: { top: (1 - heightRate) * 100 + '%' } },
        _react2.default.createElement('div', { className: 'ap-analog-clock-hand-bar',
          style: { width: width + 'px' } })
      )
    );
  }
});

exports.default = ApAnalogClockHand;
