// apeman-react-clock/shim/node/ap_analog_clock_letter.js
/**
 * A letter for analog clock.
 * @class ApAnalogClockLetter
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApAnalogClockLetter */
var ApAnalogClockLetter = _react2.default.createClass({
  displayName: 'ApAnalogClockLetter',

  propTypes: {
    angle: _react.PropTypes.number,
    letter: _react.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      angle: 0,
      letter: ''
    };
  },
  render: function render() {
    var s = this;
    var props = s.props;
    var angle = props.angle;
    var letter = props.letter;


    var containerStyle = { transform: 'rotate(' + angle + 'deg)' };
    var letterStyle = { transform: 'rotate(' + angle * -1 + 'deg)' };
    return _react2.default.createElement(
      'span',
      { className: 'ap-analog-letter',
        style: containerStyle },
      _react2.default.createElement(
        'span',
        { className: 'ap-analog-letter-title', style: letterStyle },
        letter
      )
    );
  }
});

exports.default = ApAnalogClockLetter;
