// apeman-react-clock/shim/node/ap_clock_style.js
/**
 * Style for ApClock.
 * @class ApClockStyle
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApClockStyle */
var ApClockStyle = _react2.default.createClass({
  displayName: 'ApClockStyle',

  propTypes: {
    style: _react.PropTypes.object
  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {}
    };
  },

  statics: {
    styleData: function styleData(config) {
      return {
        all: {}
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApClockStyle$styleDa = ApClockStyle.styleData(props);

    var all = _ApClockStyle$styleDa.all;
    var small = _ApClockStyle$styleDa.small;
    var medium = _ApClockStyle$styleDa.medium;
    var large = _ApClockStyle$styleDa.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApClockStyle;
