// apeman-react-clock/shim/node/ap_digital_clock_main_display.js
/**
 * Main display for digital clock
 * @class ApDigitalClockMainDisplay
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactDom = require('react-dom');

var _reactDom2 = _interopRequireDefault(_reactDom);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApDigitalClockMainDisplay */
var ApDigitalClockMainDisplay = _react2.default.createClass({
  displayName: 'ApDigitalClockMainDisplay',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    hours: _react.PropTypes.string,
    minutes: _react.PropTypes.string,
    seconds: _react.PropTypes.string,
    fontSize: _react.PropTypes.number,
    onSizeChange: _react.PropTypes.func
  },

  getInitialState: function getInitialState() {
    return {
      detailWidth: null
    };
  },
  render: function render() {
    var s = this;
    var state = s.state;
    var props = s.props;


    var fontSize = props.fontSize;

    var mainStyle = {
      fontSize: fontSize + 'px'
    },
        detailStyle = {
      fontSize: fontSize * 0.6 + 'px',
      display: 'inline-block',
      minWidth: state.detailWidth + 'px'
    };

    return _react2.default.createElement(
      'div',
      { className: 'ap-digital-clock-main-display' },
      _react2.default.createElement(
        'span',
        { className: 'ap-digital-clock-main-display-inner', ref: 'inner' },
        _react2.default.createElement(
          'span',
          { className: 'ap-digital-clock-main-label', style: mainStyle },
          _react2.default.createElement(
            'span',
            null,
            props.hours
          ),
          _react2.default.createElement(
            'span',
            null,
            ':'
          ),
          _react2.default.createElement(
            'span',
            null,
            props.minutes
          )
        ),
        _react2.default.createElement(
          'span',
          { className: 'ap-digital-clock-detail-label',
            ref: 'detail-label',
            style: detailStyle },
          _react2.default.createElement(
            'span',
            { className: 'ap-digital-clock-dummy-text',
              style: { fontSize: fontSize } },
            ' '
          ),
          _react2.default.createElement(
            'span',
            null,
            props.seconds
          )
        )
      )
    );
  },


  // --------------------
  // Lifecycle
  // --------------------

  componentWillReceiveProps: function componentWillReceiveProps() {
    var s = this;

    setTimeout(function () {
      var innerElm = _reactDom2.default.findDOMNode(s.refs['inner']);
      var detailElm = _reactDom2.default.findDOMNode(s.refs['detail-label']);
      var sizes = {
        innerWidth: innerElm.offsetWidth,
        detailWidth: detailElm.offsetWidth
      };
      s.setState(sizes);

      {
        var props = s.props;

        if (props.onSizeChange) {
          props.onSizeChange(sizes);
        }
      }
    }, 10);
  }
});

exports.default = ApDigitalClockMainDisplay;
