// apeman-react-clock/shim/node/ap_digital_clock_sub_display.js
/**
 * Sub display for digital clock
 * @class ApDigitalClockSubDisplay
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApDigitalClockSubDisplay */
var ApDigitalClockSubDisplay = _react2.default.createClass({
  displayName: 'ApDigitalClockSubDisplay',


  // --------------------
  // Specs
  // --------------------

  propTypes: {
    day: _react.PropTypes.string,
    fontSize: _react.PropTypes.number,
    width: _react.PropTypes.number
  },

  render: function render() {
    var s = this;
    var props = s.props;
    var fontSize = props.fontSize;
    var width = props.width;


    var displayStyle = {
      fontSize: fontSize + 'px'
    },
        labelStyle = {
      width: width + 'px'
    };

    return _react2.default.createElement(
      'div',
      { className: 'ap-digital-clock-sub-display', style: displayStyle },
      _react2.default.createElement(
        'div',
        { className: 'ap-digital-clock-sub-label', style: labelStyle },
        _react2.default.createElement(
          'span',
          null,
          props.day
        )
      )
    );
  }
});

exports.default = ApDigitalClockSubDisplay;
