// apeman-react-clock/shim/node/ap_digital_clock_style.js
/**
 * Style for ApDigitalClock
 * @class ApDigitalClock
 */

'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _apemanReactStyle = require('apeman-react-style');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/** @lends ApDigitalClock */
var ApDigitalClock = _react2.default.createClass({
  displayName: 'ApDigitalClock',

  propTypes: {
    style: _react.PropTypes.object,
    width: _react.PropTypes.number,
    height: _react.PropTypes.number,
    color: _react.PropTypes.string,
    backgroundColor: _react.PropTypes.string

  },
  getDefaultProps: function getDefaultProps() {
    return {
      style: {},
      width: 340,
      height: 240,
      color: '#333',
      backgroundColor: '#FFF'
    };
  },

  statics: {
    styleData: function styleData(config) {
      var width = config.width;
      var height = config.height;
      var color = config.color;
      var backgroundColor = config.backgroundColor;

      return {
        all: {
          '.ap-digital-clock': {
            color: color,
            width: width,
            height: height,
            display: 'block',
            margin: '0 auto',
            textAlign: 'center'
          },
          '.ap-digital-clock-board': {
            backgroundColor: backgroundColor,
            display: 'inline-block',
            border: '2px solid ' + color,
            position: 'relative',
            background: '' + backgroundColor
          },
          '.ap-digital-clock-board-inner': {
            display: 'inline-block',
            verticalAlign: 'middle'
          },
          '.ap-digital-clock-dummy-text': {
            opacity: 0,
            width: 1,
            display: 'inline-block',
            verticalAlign: 'middle',
            marginRight: '-1px',
            whiteSpace: 'pre',
            overflow: 'hidden'
          },
          '.ap-digital-clock-main-display': {},
          '.ap-digital-clock-sub-display': {
            overflow: 'hidden',
            textAlign: 'right'
          },
          '.ap-digital-clock-sub-label': {
            display: 'block',
            margin: '2px auto'
          }
        }
      };
    }
  },
  render: function render() {
    var s = this;
    var props = s.props;

    var _ApDigitalClock$style = ApDigitalClock.styleData(props);

    var all = _ApDigitalClock$style.all;
    var small = _ApDigitalClock$style.small;
    var medium = _ApDigitalClock$style.medium;
    var large = _ApDigitalClock$style.large;


    return _react2.default.createElement(
      _apemanReactStyle.ApStyle,
      { data: Object.assign(all, props.style),
        smallMediaData: small,
        mediumMediaData: medium,
        largeMediaData: large
      },
      props.children
    );
  }
});

exports.default = ApDigitalClock;
